src/store/modules/agent.js

```diff
@@ -223,9 +237,66 @@ export default ({
     },
     setIsDefaultViewOnAgent (state, value) {
       state.isDefaultViewOnAgent = value
+    },
+    updateAuthenticationToken (state, token) {
+      state.authentication_token = token
     }
   },
   actions: {
+    login ({ commit, state }, credentials) {
```

Comment:
we can move the login action to a file src/store/modules/auth.js instead in the agent.js